### PR TITLE
Limit 'size' parameter to 1000 or less (#1257)

### PR DESCRIPTION
**Pull request by achave11**

### CODE REVIEW

File: lambdas/service/app.py [hunk 1]
```diff
@@ -194,12 +194,30 @@ def validate_repository_search(params, **validators):
...
+    ...
+    chalice.app.BadRequestError: BadRequestError: Invalid size value, must be less than 1000
+    """
+    try:
+        size = int(size)
+    except:
+        raise BadRequestError(f'Invalid type parameter for `size`')
+    else:
+        if size > 1000:
+            raise BadRequestError(f'Invalid size value, must be less than 1000')
```
Comment (danielsotirhos) [inline lambdas/service/app.py]:
```suggestion
            raise BadRequestError(f'Invalid size value, must be less than 1000')
        elif size < 1:
            raise BadRequestError(f'Invalid size value, must be greater than 0')
```

File: lambdas/service/app.py [hunk 2]
```diff
@@ -195,12 +195,32 @@ def validate_repository_search(params, **validators):
...
+
+    >>> validate_size('100000')
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: Invalid size value, must be less than 1000
+    """
+    try:
+        size = int(size)
+    except:
+        raise BadRequestError(f'Invalid type parameter for `size`')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
        raise BadRequestError(f'Invalid value for parameter `size`')
```

File: lambdas/service/app.py [hunk 3]
```diff
@@ -195,12 +195,32 @@ def validate_repository_search(params, **validators):
...
+    else:
+        if size > 1000:
+            raise BadRequestError(f'Invalid size value, must be less than 1000')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
            raise BadRequestError(f'Invalid value for parameter `size`, must not be greater than {max_size}')
```

File: lambdas/service/app.py [hunk 4]
```diff
@@ -195,12 +195,32 @@ def validate_repository_search(params, **validators):
...
+        elif size < 1:
+            raise BadRequestError(f'Invalid size value, must be greater than 0')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
            raise BadRequestError(f'Invalid value for parameter `size`, must be greater than 0')
```

File: lambdas/service/app.py [hunk 5]
```diff
@@ -195,12 +195,32 @@ def validate_repository_search(params, **validators):
...
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: Invalid size value, must be less than 1000
+    """
+    try:
+        size = int(size)
+    except:
+        raise BadRequestError(f'Invalid type parameter for `size`')
+    else:
+        if size > 1000:
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
        max_size = 1000
        if size > max_size:
```

File: lambdas/service/app.py [hunk 6]
```diff
@@ -213,12 +213,33 @@ def validate_repository_search(params, **validators):
...
+
+    >>> validate_size('100000')
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: Invalid size value, must be less than 1000
+    """
+    try:
+        size = int(size)
+    except:
+        raise BadRequestError(f'Invalid value for parameter for `size`')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
        raise BadRequestError(f'Invalid value for parameter `size`')
```

File: lambdas/service/app.py [hunk 7]
```diff
@@ -213,12 +213,33 @@ def validate_repository_search(params, **validators):
...
 
 
+def validate_size(size):
+    """
+    >>> validate_size('999')
+
+    >>> validate_size('100000')
+    Traceback (most recent call last):
+    ...
+    chalice.app.BadRequestError: BadRequestError: Invalid size value, must be less than 1000
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]: I'd add coverage for the remaining cases, too.

https://coveralls.io/builds/26049933/source?filename=lambdas%2Fservice%2Fapp.py#L219
https://coveralls.io/builds/26049933/source?filename=lambdas%2Fservice%2Fapp.py#L221

File: lambdas/service/app.py [hunk 8]
```diff
@@ -213,12 +213,41 @@ def validate_repository_search(params, **validators):
...
-        'size': int,
+        'size': validate_size,
         'sort': validate_facet,
         **validators
     })
 
 
+def validate_size(size):
+    """
+    >>> validate_size('999')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
    >>> validate_size('1000')
```

File: lambdas/service/app.py [hunk 9]
```diff
@@ -213,12 +213,41 @@ def validate_repository_search(params, **validators):
...
+
+    >>> validate_size('100000')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py]:
```suggestion
    >>> validate_size('1001')
```

**Comment by achave11** [comment]
@hannes-ucsc apologies for the confusion. 
https://github.com/databiosphere/azul/issues/1257 specified disallowing any value above 999. That is why some of the test and logic where a bit fuzzy.

**Pull request merged**